test/lambda/mock/transaction.js

```diff
@@ -14,6 +15,23 @@ module.exports = class TransactionMock {
     Object.assign(this.customContext, custom)
   }
 
+  setCloudContext (cloud) {
+    Object.assign(this._cloud = cloud)
+  }
+
+  setMessageContext (message) {
+    Object.assign(this._message = message)
+  }
+
+  setServiceContext (serviceContext) {
+    if (!serviceContext) return
+    this._service = Object.assign(this._service || {}, serviceContext)
+  }
+
+  setFaas (context) {
+    Object.assign(this._faas, context)
```

Comment:
The impl of setCloudContext, setMessageContext, setFaas are wrong here -- though are working by luck.   You perhaps (tho not required) want `this._cloud = undefined` et al in the constructor; then you want the implementations to be like `setServiceContext`.

Reply:
ab9a381164daa60f93783a8935387c841f872bab